include/algorithms/synthesis/syrec_synthesis_additional_lines.hpp

```diff
@@ -14,8 +14,35 @@ namespace syrec {
             return !SyrecSynthesis::on_statement(statement);
         }
 
-        bool on_statement(const assign_statement& statement) override;
+        bool op_rhs_lhs_expression([[maybe_unused]] const expression::ptr& expression, [[maybe_unused]] std::vector<unsigned>& v) override {
+            return true;
+        }
+        bool op_rhs_lhs_expression([[maybe_unused]] const variable_expression& expression, [[maybe_unused]] std::vector<unsigned>& v) override {
+            return true;
+        }
+        bool op_rhs_lhs_expression([[maybe_unused]] const binary_expression& expression, [[maybe_unused]] std::vector<unsigned>& v) override {
+            return true;
+        }
```

Comment:
instead of defining trivial functions here, consider moving these definitions to the base class and just remove the functions here.
This way, there is a trivial default implementation in the base class and the synthesis approach without additional lines can provide appropriate overheads.

Reply:
done 👍🏼